lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -130,6 +130,27 @@ def set_3d_properties(self, zs=0, zdir='z'):
         self._verts3d = juggle_axes(xs, ys, zs, zdir)
         self.stale = True
 
+    def set_data_3d(self, *args):
+        """
+        Set the x, y and z data
+
+        ACCEPTS: 3D array (rows are x, y, z) or three 1D arrays
+        """
+        if len(args) == 1:
+            x, y, z = args[0]
+        else:
+            x, y, z = args
+
+        self.set_xdata(x)
+        self.set_ydata(y)
+        self.set_3d_properties(zs=z)
+
+    def get_data_3d(self):
+        """
+        Return the xdata, ydata, zdata.
+        """
+        return self._verts3d
```

Comment:
Does this need to un-juggle the order of the data?

Reply:
It probably should. We're calling `set_3d_properties` with no `zdir` so we shouldn't do any juggling but the user could change the zdir, I'll probably also allow an optional kwarg to allow the user to also set zdir right from set_data_3d.

Reply:
Both should probably get the `zdir` kwarg?

Reply:
The juggle axis feature is for converting a 2D object into 3D. It essentially states which dimension is going to be a constant. It is not applicable here.